### PR TITLE
Strange fix to make functests work under valgrind in jenkins

### PR DESCRIPTION
**Pull request by kzangeli**

### CODE REVIEW

**Comment by iariasleon** [comment]
LGTM

File: test/functionalTest/harnessFunctions.sh [hunk 1]
```diff
@@ -278,6 +278,7 @@ function brokerStartAwait
     if [ "$?" == "0" ]
     then
       vMsg The orion context broker has started, listening on port $port
+      echo "Broker started after $loopNo checks" >> /tmp/brokerStart
```
Comment (kzangeli) [inline test/functionalTest/harnessFunctions.sh]: This here is the list that 'fixes' the problem.
Hard to believe, right?

Also, even if the echo is done to /dev/null, the 'fix' still works.
Now, at least some info is gathered in /tmp/brokerStart (emptied in test/functionalTest/testHarness.sh)

NTC

Comment (fgalan) [inline test/functionalTest/harnessFunctions.sh]: A comment should be added above this line explaining that if this line is removed then problems may occur due to unknown causes. Otherwise, we could forgot about this fix and, passed some time, removed this appartently "harmless" echo line.

File: test/functionalTest/harnessFunctions.sh [hunk 2]
```diff
@@ -289,9 +290,10 @@ function brokerStartAwait
 
   sleep .5
 
-  # Check that CB started fine
-  curl -s localhost:${port}/version | grep version > /dev/null
+  # Check that CB started
+  curl -s localhost:${port}/version | grep version >> /tmp/brokerStart
```
Comment (fgalan) [inline test/functionalTest/harnessFunctions.sh]: Is this related with the fix?

File: test/functionalTest/testHarness.sh [hunk 3]
```diff
@@ -25,6 +25,7 @@
 date
 testStartTime=$(date +%s.%2N)
 MAX_TRIES=${CB_MAX_TRIES:-3}
+echo $testStartTime > /tmp/brokerStart
```
Comment (fgalan) [inline test/functionalTest/testHarness.sh]: Is this related with the fix?

Comment (kzangeli) [inline test/functionalTest/testHarness.sh]: Yes, I need to remove the file /tmp/brokerStart somewhere, as only '>>' is used.
If I don't do that, the file would grow and grow and grow ...

File: test/functionalTest/harnessFunctions.sh (hunk 2 above)
Comment (kzangeli) [inline test/functionalTest/harnessFunctions.sh]: Yes, just some more info in /tmp/brokerStart now that we have the file ...
Without this change I would have had to add some other line to indicate that the broker actually responds.

File: test/functionalTest/harnessFunctions.sh (hunk 1 above)
Comment (kzangeli) [inline test/functionalTest/harnessFunctions.sh]: True, it would be good to have this strange thing explained

File: test/functionalTest/harnessFunctions.sh (hunk 2 above)
Comment (fgalan) [inline test/functionalTest/harnessFunctions.sh]: I'd suggest to rename the file to something more meanifull taking into accont its content, e.g.

/tmp/brokerStart.counter
